### ISSUE TITLE
ReferenceError: dragula is not defined

### ISSUE THREAD

**Issue by thecyberbob** [rfox90/meteor-dragula#7]
I followed your instructions in my app setting up a template like so

```
<template name="dragList">
  <ul>
    <div class='wrapper'>
    <div id='left1' class='container'>
    {{#each tasks}}
      <div>{{> task}}</div>
    {{/each}}
    </div>
  </div>
  </ul>
</template>
```

And the onRendered like so

```
Template.dragList.onRendered(function(){
  dragula([document.getElementById('left1')]);
});
```

But I get the dragula is not defined on the middle line of the Template.dragList.onRendered. I'm running Meteor version 1.2.1

Any ideas?

**Comment by benkelaar** [comment]
I have the same problem, but the dragula function is defined in my window scope. (I can call it from the chrome console)

The example works just fine. I first thought that maybe it had something to the with the fact that my template is in the client folder instead of in the top folder but I reworked the example into that and that works as well. Puzzled.

Hmm. By now the problem has miraculously resolved itself on my side. It seemed to be fixed when I added a container class to my drag targets, but when I removed those it still worked. May have been that client side dependency wasn't properly updated or something. But at least it works for me now. Maybe my story can help you, sorry I couldn't be more specific.

**Comment by thecyberbob** [comment]
Whew. That is odd. I wonder if it's a load order thing? Maybe I'll try calling the library in the header directly and see if that wakes it up. Thanks for the support though. At least I'm not alone.

**Comment by benkelaar** [comment]
I had it a second time, now I wrapped a Meteor.defer() around it and that seems to fix it. Kinda hacky but I'll take it. I guess it is dependent on load order. Removing the dragula call from the page render flow may just be the workaround we need.

**Comment by rfox90** [comment]
Hi Sorry for the delay here,

Dragula depends on a weird loading order. The document body has to be loaded before the dragula code runs. 

I'll spinup the example project and see if it has this problem. Is there any chance you can isolate it to a reproduceable repo.

I can't wait for meteor 1.3.

**Comment by thecyberbob** [comment]
Thanks for the reply!

I could isolate it somewhat. But there really isn't much to what I have. The app is very similar to the tutorial app from Meteor with the container surrounding the part where it prints off of the task items.

Regardless I'll see what I can do.

1.3 has a better load order control I take it?

**Comment by rfox90** [comment]
1.3 has native support for NPM this package will be redundant.
I'll check out the tutorial app asap

**Comment by benkelaar** [comment]
I couldn't isolate it either. I also tried and failed to reproduce it in the example app. Locally Meteor.defer fixed it (And this also fixed the error in the console) but when running on the meteor server it then just doesn't initialize at all. Added a button to initialize after the fact as a workaround for now.

Source code for the project I have it in can be found here: https://github.com/benkelaar/java-knowledge.

Maybe that helps, I've tried various ways of initializing dragula, sometimes it gives the error, sometimes not, also sometimes varying with what should be the same setup process. So I'm kind of puzzled.

As a workaround for this project I'm probably going to add a "Set editable" button to the project which initializes dragula amongst other things.
